apps/finance/contracts/Finance.sol

```diff
@@ -139,14 +133,14 @@ contract Finance is AragonApp, ERC677Receiver {
     * @param _amount Amount of tokens sent
     * @param _reference Reason for payment
     */
-    function deposit(ERC20 _token, uint256 _amount, string _reference) external transitionsPeriod {
+    function deposit(address _token, uint256 _amount, string _reference) external transitionsPeriod {
         _recordIncomingTransaction(
             _token,
             msg.sender,
             _amount,
             _reference
         );
-        require(_token.transferFrom(msg.sender, address(vault), _amount));
+        vault.deposit(_token, msg.sender, _amount, new bytes(0));
```

Comment:
So now before depositing tokens using Finance you need to create an allowance to the Vault? (otherwise there is no way this works haha). This should be made very clear in the documentation. Also pinging @onbjerg and @sohkai  who are working in the frontend and this could be a blocker

Reply:
Yes, this is not very user friendly, I'm changing it to get the tokens to finance app, allow to Vault and then deposit, in the way `receiveApproval` was done.

Reply:
Hmm, I think this was always the case, to be honest. We could get away with it in one call if the token had `approveAndCall` / ERC677, but I guess that's not in here anymore.

Reply:
Yes, but what was weird (from UX perspective) was to approve to `Vault` but to send to `Finance`. Have a look at how it is now, please, and let me know your thoughts.